src/Tests/Dns.fs

```diff
@@ -306,4 +311,122 @@ let tests = testList "DNS Zone" [
             } |> ignore
         ) "Should fail when add_nsd_reference was already called"
     }
+    test "Private DNS Zone is created with records" {
+        let resources =
+            arm {
+                add_resources [
+                    dnsZone {
+                        name "farmer.com"
+                        zone_type Private
+                        add_records [
+                            cnameRecord {
+                                name "www"
+                                ttl 3600
+                                cname "farmer.com"
+                                zone_type Private
```

Comment:
when adding a record to a zone with `add_records`, should the zone type be automatically set to that of the zone?
I presume it's not possible to add a public record to a private zone or vice versa